kong/plugins/aws-request-signing/handler.lua

```diff
@@ -112,46 +93,45 @@ if _TEST then
   AWSLambdaSTS._get_iam_credentials = get_iam_credentials
 end
 
-function AWSLambdaSTS.access(_self, conf)
+function AWSLambdaSTS:access(conf)
```

Comment:
As @DanielRailean mentioned in his review, we actually changed this to use `_self` instead of sending it with `:access` because the linter doesn't seem to like `:access`.

We are by no means experts in Lua though, so I'm very eager to hear if we are doing something bad with this approach?
Maybe the linter is wrong and we should fix that instead of the code?
If we can learn something new here, I would be very happy.

Reply:
Sorry for not running `luacheck` on the code myself - The missing `local` on the `upstream_headers` variable was not necessary.

I've added a specific `ignore` rule to the `.luacheck` file that prevents luacheck from complaining about unused `self` arguments.  We're doing this in our codebase in several places and it seems to be better than cluttering the code with an extra `_self` argument.  This is more of a style choice, of course.